src/callBuilder.js

```diff
@@ -0,0 +1,139 @@
+        this.checkFilter();
+        var promise = this._sendNormalRequest(this.url)
+            .then(this._parseResponse.bind(this));
+        return promise;
+    }
+
+    /**
+    * Convert each link into a function on the response object.
+    */
+    _parseRecord(json) {
+        if (!json._links) {
+            return json;
+        }
+        var self = this;
+        var linkFn = function (link) {
+            return function (opts) {
+                if (link.template) {
+                    let template = URITemplate(link.href);
+                    return self._sendNormalRequest(URI(template.expand(opts)));
+                } else {
+                    return self._sendNormalRequest(URI(link.href));
+                }
+            };
+        };
+        Object.keys(json._links).map(function(value, index) {
+            var link = json._links[value];
+            json[value] = linkFn(link);
+        });
+        return json;
+    }
+    
+    _sendNormalRequest(url) {
+        // To fix:  #15 Connection Stalled when making multiple requests to the same resource
+        url.addQuery('c', Math.random());
+        var promise = axios.get(url.toString())
+            .then(response => response.data)
+            .catch(this._handleNetworkError);
+        return toBluebird(promise);
+    }
+
+    _parseResponse(json) {
+        if (json._embedded && json._embedded.records) {
+            return this._toCollectionPage(json);
+        } else {
+            return this._parseRecord(json);
+        }
+    }
+
+    _toCollectionPage(json) {
+        var self = this;
+        for (var i = 0; i < json._embedded.records.length; i++) {
+            json._embedded.records[i] = this._parseRecord(json._embedded.records[i]);
+        }
+        return {
+            records: json._embedded.records,
+            next: function () {
+                return self._sendNormalRequest(URI(json._links.next.href))
+                    .then(self._toCollectionPage.bind(self));
+            },
+            prev: function () {
+                return self._sendNormalRequest(URI(json._links.prev.href))
+                    .then(self._toCollectionPage.bind(self));
+            }
+        };
+    }
+
+    _handleNetworkError(response) {
+        if (response instanceof Error) {
+            return Promise.reject(response);
+        } else {
+            switch (response.status) {
+                case 404:
+                    return Promise.reject(new NotFoundError(response.data, response));
+                default:
+                    return Promise.reject(new NetworkError(response.status, response));
+            }
+        }
+    }
+
+    after(token) {
+        this.url.addQuery("after", token);
+        return this;
+    }
+
+    limit(number) {
+        this.url.addQuery("limit", number);
+        return this;
+    }
+
+    order(direction) {
+        this.url.addQuery("order", direction);
+        return this;
+    }
+
+    stream(options) {
```

Comment:
This is another important method on this class.  Having some documentation about its role within the system would be great.